File: Recursion/MaxElementInArray/minMax.cpp
#include <vector>
#include <climits>
using namespace std;

class Solution {
  public:
    int findMax(vector<int> &arr,int idx){
        if(arr.size()==idx)return INT_MIN;
        return max(arr[idx],findMax(arr,idx+1));
    }
    int findMin(vector<int> &arr,int idx){
        if(arr.size()==idx)return INT_MAX;
        return min(arr[idx],findMin(arr,idx+1));
    }
    vector<int> getMinMax(vector<int> &arr) {
        // code here()
        vector<int>ans(2,0);
        ans[0] = findMin(arr,0);
        ans[1] = findMax(arr,0);
        return ans;
    }
};

File: Recursion/MaxElementInArray/minMax.py
import sys

class Solution:
    def findMax(self,arr,idx):
        if len(arr) == idx: return sys.maxsize* -1
        return max(arr[idx],self.findMax(arr,idx+1))
        
        
    def findMin(self,arr,idx):
        if len(arr) == idx: return sys.maxsize* 1
        return min(arr[idx],self.findMin(arr,idx+1))      
        

    def getMinMax(self, arr):
        # code here
        ans = [0]*2
        ans[0] = self.findMin(arr,0)
        ans[1] = self.findMax(arr,0) 
        return ans